src/garage/_primitives.py

```diff
@@ -0,0 +1,152 @@
+"""Policy interface."""
+import abc
+
+
+class Primitive(abc.ABC):
+    """Primitive interface.
+
+    Args:
+        name (str): Name for logs, variable scopes, etc.
+        spec (garage.InOutSpec): Input/output specification this primitive will
+            satisfy.
+
+    """
+
+    def __init__(self, name, spec):
+        self._name = name
+        self._spec = spec
+
+    @abc.abstractmethod
+    def get_parameters(self):
+        """Get parameters of this primitive.
+
+        Returns:
+            dict[str, np.ndarray]: A dictionary mapping from names to values of
+                each parameter.
+
+        """
+
+    @abc.abstractmethod
+    def set_parameters(self, parameters):
+        """Set the parameters of this primtive.
+
+        Args:
+            parameters (dict[str, np.ndarray]): A dictionary mapping from names
+                to values of each parameter.
+
+        """
+
+    @property
+    def spec(self):
+        """InOutSpec: The input/output spec for this primitive."""
+        return self._spec
+
+    @property
+    def name(self):
+        """str: Name of this primitive."""
+        return self._name
+
+    @abc.abstractmethod
+    def clone(self, name):
+        """Create a fully-independent copy of this primitive.
+
+        Args:
+            name (str): Name for the cloned instance.
+
+        Returns:
+            Primitive: A fully-independent copy of this primitive.
+
+        """
+
+    @abc.abstractmethod
+    def __getstate__(self):
+        """See Object.__getstate__.
+
+        Returns:
+            dict: Pickling state for the primitive
+
+        """
+
+    @abc.abstractmethod
+    def __setstate__(self, state):
+        """See Object.__setstate__.
+
+        Args:
+            state (dict): Unpickled state of this primitive.
+
+        """
+
+
+class Policy(Primitive):
```

Comment:
On the naming of `Primitive`. I have a feeling that a policy is a large thing that does not fit well with the nuance of the smallness of Primitive. What do you think of the name `Module`?

Reply:
`Module` is already taken by a very specific abstraction in tf and torch, but I am thinking we should rename this Model.

Reply:
is this `Model` related to the `Model` defined in TF branch?

Reply:
this would replace `garage.tf.Model` which would be partially- or fully-replaced by `tf.Module`

A tf policy would inherit from `tf.Module` and implement `garage.Policy`, but (as KR has discussed), it may not actually inherit from `garage.Policy`.